node.gyp

```diff
@@ -759,12 +824,44 @@
         } ],
       ]
     },
+    {
+      # When using shared lib to build executable in Windows, in order to avoid
+      # filename collision, the executable is node-win.exe. Need to rename it
+      # back to node.exe
+      'target_name': 'rename_node_bin_win',
+      'type': 'none',
+      'dependencies': [
+        '<(node_core_target_name)',
+      ],
+      'conditions': [
+        [ 'OS=="win" and node_intermediate_lib_type=="shared_library"', {
+          'actions': [
+            {
+              'action_name': 'rename_node_bin_win',
+              'inputs': [
+                '<(PRODUCT_DIR)/<(node_core_target_name)-win.exe'
+              ],
+              'outputs': [
+                '<(PRODUCT_DIR)/<(node_core_target_name).exe',
+              ],
+              'action': [
+                'mv', '<@(_inputs)', '<@(_outputs)',
+              ],
+            },
+          ],
+        } ],
+      ]
+    },
     {
       'target_name': 'cctest',
       'type': 'executable',
 
       'dependencies': [
+        # This can't depend on node_lib_target_name, the reasoning be
+        # is the shared lib in Windows only exports NODE_EXTERN functions
+        # and cctest needs some internal functions which are not NODE_EXTERN.
```

Comment:
Maybe we should split cctest into internal and external tests, that way we could test the shared library on the external `cctest` tests.

Reply:
good point. I think it depends on the value of running cctest against the shared lib. If it's good and necessary, then we should do it.